### PR TITLE
OCM-2636: Enable importing machine pools

### PR DESCRIPTION
**Pull request by JohnStrunk**
Tested importing:
- 1AZ cluster, 1AZ pool
- mAZ cluster, 1AZ pool
- mAZ cluster, mAZ pool

Note: I removed the verification that replicas is a multiple of 3 in the case of mAZ pools. OCM provides this validation separately. The reason for the removal from the TF code is that it was causing a problem during updates after state import wrt the 1AZ optional fields for mAZ clusters. It would have required additional API queries to read the cluster object.

### CODE REVIEW

File: provider/machine_pool_resource.go [hunk 1]
```diff
@@ -455,34 +456,11 @@ func (r *MachinePoolResource) Update(ctx context.Context, request tfsdk.UpdateRe
...
-		)
-		return
-	}
-
 	computeNodesEnabled := false
 	autoscalingEnabled := false
 
 	if !plan.Replicas.Unknown && !plan.Replicas.Null {
 		computeNodesEnabled = true
-		if isMultiAZPool && plan.Replicas.Value%3 != 0 {
```
Comment (nirarg) [inline provider/machine_pool_resource.go:476]: Why did you remove this validation?
In case of multiple AZ, replicas must be replica of 3
This is the relevant validation in the cli:
https://github.com/openshift/rosa/blob/4de5ba8ee9b6b9eff715d3f8c5fa9ec362047584/cmd/edit/machinepool/machinepool.go#L65

Comment (JohnStrunk) [reply provider/machine_pool_resource.go:476]: I'm the one that added the check originally: https://github.com/terraform-redhat/terraform-provider-rhcs/commit/bc3214e95efa633429ac62afceb5a4a9f80cd00a
OCM validates the replica count, so it isn't actually necessary... I just added it because (1) it was in the cli and (2) I had the info for other reasons.

I removed it because it relies on knowing `isMultiAZPool` which comes from `validateAZConfig()`, and the logic in the validate function wasn't able to cope w/ the state from import. It comes down to the validation using the user config values. However, import (Read) doesn't set them in the state, so TF believes the fields are null. Since they are locked from updating, the user can't set them (nor should they need to, honestly, since the values and validation only matter on Create, not Update).

I add a new function to determine the multi-ness of the pool and keep this validation if you'd like.

Comment (nirarg) [reply provider/machine_pool_resource.go:476]: But the validateAZConfig use the plan and not the state, so maybe it will work with the state?
